### DISCUSSION TITLE
Creating user-defined commands

### DISCUSSION THREAD

**Discussion by hamyyy** [terjeio/grblHAL#150] [Q&A] [answered]
Does grblHAL have user-defined commands implemented yet? I've seen a few comments int the source code (`ModalGroup_M10`) but not sure if that's a thing yet. if it is, can someone point me in the right direction to set up a custom G-code command? Thanks :)

**Comment by terjeio** [comment, marked as answer]
Take a look at [this code](https://github.com/terjeio/grblHAL/blob/test/templates/mcodes.c) and [this](https://github.com/terjeio/grblHAL/blob/test/plugins/laser/coolant.c) too.

Note that only custom M-codes are supported, not G-codes.
